ports/samd/mpconfigport.h

```diff
@@ -123,7 +125,7 @@ __attribute__((always_inline)) static inline uint32_t disable_irq(void) {
     do { \
         extern void mp_handle_pending(bool); \
         mp_handle_pending(true); \
-        __WFI(); \
+        __WFE(); \
```

Comment:
I'm not sure why this works, because there are no SEV calls anywhere in this port, so WFE can never be woken.

I would prefer to use WFE rather than WFI because you have more control over when to wake the CPU, and it's easier to avoid race conditions.  Using WFI you usually need to do this:
```c
void wait(void) {
    for(;;) {
        disable_irqs;
        if (event/interrupt of interest occurred) {
            enable_irqs;
            return;
        }
        WFI;
        enable_irqs;
    }
}
```

WFE on its own can effectively replace the above function.

Reply:
As far as I understood it, WFE also responds to a hardware interrupt, and maybe other hardware events, like from the . The difference here is, that with WFI the SPI unit wait for the next Systick interrupt after the last byte has been transferred, but using WFE it finishes immediately. That's why I replaced WFI by WFE.
Both instructions result in the same power consumption in REPL.

Reply:
OK, I guess the SAMD architecture has its own way of handling events/interrupts (I think the MCU can decide how to handle them, it's not part of the ARM Cortex spec, although I could be wrong there).